columnflow/production/cms/pdf.py

```diff
@@ -30,24 +30,55 @@
     # only run on mc
     mc_only=True,
 )
-def pdf_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
+def pdf_weights(
+    self: Producer,
+    events: ak.Array,
+    outlier_threshold: float = 0.5,
+    outlier_action: str = "ignore",
+    outlier_log_mode: str = "warning",
+    **kwargs,
+) -> ak.Array:
     """
     Producer that determines the pdf up and down variations on an event-by-event basis.
     This producer assumes that the nominal entry is always the first LHEPdfWeight value and
     that the nominal weight is already included in the LHEWeight.
     Can only be called with MC datasets.
 
+    The *outlier_action* defines the procedure of how to handle events with a pdf
+    uncertainty above the *outlier_threshold*. Supported modes are:
+
+        - ``"ignore"``: events are kept unmodified
+        - ``"remove"``: pdf weight nominal/up/down are all set to 0
+        - ``"raise"``: an exception is raised
+
+     Additionally, the verbosity of the procedure can be set with *outlier_log_mode*,
+     which offers the following options:
+
+        - ``"none"``: no message is given
+        - ``"info"``: a `logger.info` message is given
+        - ``"debug"``: a `logger.debug` message is given
+        - ``"warning"``: a `logger.warning` message is given
+
     Resources:
 
        - https://arxiv.org/pdf/1510.03865.pdf
     """
+
+    if not outlier_action:
+        outlier_action = "ignore"
+    if not outlier_log_mode:
+        outlier_log_mode = "none"
+
+    assert outlier_action in ("ignore", "remove", "raise")
+    assert outlier_log_mode in ("none", "info", "debug", "warning")
```

Comment:
Just some last thing: we should remove all `assert`'s from our code base (as done 2a76125).

```suggestion
    if outlier_action not in (known_actions := ("ignore", "remove", "raise")):
        raise ValueError(
            f"unknown outlier_action '{outlier_action}', known values are {','.join(known)}",
        )
    if outlier_log_mode not in (known_log_modes := ("none", "info", "debug", "warning")):
        raise ValueError(
            f"unknown outlier_log_mode '{outlier_log_mode}', known values are "
            f"{','.join(known_log_modes)}",
        )
```